crates/bevy_ecs/src/schedule/system_set.rs

```diff
@@ -145,4 +165,36 @@ impl SystemSet {
         }
         (run_criteria, systems)
     }
+
+    fn sequentialize(systems: &mut Vec<SystemDescriptor>) {
+        let start = NEXT_SEQUENCE_ID.fetch_add(systems.len() as u32, Ordering::Relaxed);
+        let mut last_label: Option<SequenceId> = None;
+        for (idx, descriptor) in systems.iter_mut().enumerate() {
+            let label = SequenceId(start + idx as u32);
+            match descriptor {
+                SystemDescriptor::Parallel(descriptor) => {
+                    descriptor.labels.push(label.dyn_clone());
+                    if let Some(ref after) = last_label {
+                        descriptor.after.push(after.dyn_clone());
+                    }
+                }
+                SystemDescriptor::Exclusive(descriptor) => {
+                    descriptor.labels.push(label.dyn_clone());
+                    if let Some(ref after) = last_label {
+                        descriptor.after.push(after.dyn_clone());
+                    }
+                }
+            }
+            last_label = Some(label);
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+struct SequenceId(u32);
+
+impl SystemLabel for SequenceId {
+    fn dyn_clone(&self) -> Box<dyn SystemLabel> {
+        Box::new(<SequenceId>::clone(self))
+    }
```

Comment:
Any reason to not just
```rust
#[derive(SystemLabel)]
```
instead of manually implementing it?

Reply:
I get this error when using it:
```
use of undeclared crate or module `bevy_ecs`
```

Seems like the derive macro doesn't work within bevy_ecs itself.

Reply:
Ugh, that's annoying :(